nntrainer/include/layer.h

```diff
@@ -388,7 +324,31 @@ class Layer {
     return weight_list.get()[position];
   }
 
+#if defined(ENABLE_TEST)
+  /**
+   * @brief Set the batch for the layer
+   * @param batch Batch value to be set
+   * @todo Make this private. Only model should be able to do this.
+   */
+  void setBatch(unsigned int batch) {
```

Comment:
I dont like this either. Please let me know if you have any suggestions for this.